stan/math/fwd/fun/multiply.hpp

```diff
@@ -10,120 +11,50 @@
 namespace stan {
 namespace math {
 
-template <typename T, int R1, int C1>
-inline Eigen::Matrix<fvar<T>, R1, C1> multiply(
-    const Eigen::Matrix<fvar<T>, R1, C1>& m, const fvar<T>& c) {
-  Eigen::Matrix<fvar<T>, R1, C1> res(m.rows(), m.cols());
-  for (int i = 0; i < m.size(); i++)
-    res(i) = c * m(i);
-  return res;
-}
-
-template <typename T, int R2, int C2>
-inline Eigen::Matrix<fvar<T>, R2, C2> multiply(
-    const Eigen::Matrix<fvar<T>, R2, C2>& m, double c) {
-  Eigen::Matrix<fvar<T>, R2, C2> res(m.rows(), m.cols());
-  for (int i = 0; i < m.size(); i++)
-    res(i) = c * m(i);
-  return res;
-}
-
-template <typename T, int R1, int C1>
-inline Eigen::Matrix<fvar<T>, R1, C1> multiply(
-    const Eigen::Matrix<double, R1, C1>& m, const fvar<T>& c) {
-  Eigen::Matrix<fvar<T>, R1, C1> res(m.rows(), m.cols());
-  for (int i = 0; i < m.size(); i++)
-    res(i) = c * m(i);
-  return res;
-}
-
-template <typename T, int R1, int C1>
-inline Eigen::Matrix<fvar<T>, R1, C1> multiply(
-    const fvar<T>& c, const Eigen::Matrix<fvar<T>, R1, C1>& m) {
-  return multiply(m, c);
-}
-
-template <typename T, int R1, int C1>
-inline Eigen::Matrix<fvar<T>, R1, C1> multiply(
-    double c, const Eigen::Matrix<fvar<T>, R1, C1>& m) {
-  return multiply(m, c);
-}
-
-template <typename T, int R1, int C1>
-inline Eigen::Matrix<fvar<T>, R1, C1> multiply(
-    const fvar<T>& c, const Eigen::Matrix<double, R1, C1>& m) {
-  return multiply(m, c);
-}
-
-template <typename T, int R1, int C1, int R2, int C2>
-inline Eigen::Matrix<fvar<T>, R1, C2> multiply(
-    const Eigen::Matrix<fvar<T>, R1, C1>& m1,
-    const Eigen::Matrix<fvar<T>, R2, C2>& m2) {
+template <typename T1, typename T2,
+          typename = require_all_eigen_vt<is_fvar, T1, T2>,
+          typename = require_same_vt<T1, T2>,
+          typename = require_not_eigen_row_and_col_t<T1, T2>, unsigned int = 0>
```

Comment:
Also why is unsigned int here?

Reply:
see https://github.com/stan-dev/math/pull/1628#discussion_r368457618